features/step_definitions/files_steps.rb

```diff
@@ -1,5 +1,10 @@
+Given(/^I have an uncommitted file$/) do
+  @uncommitted_file_name = create_uncommitted_file
+end
+
+
 Given(/^I have an uncommitted file with name: "(.+?)" and content: "(.+?)"$/) do |name, content|
-  IO.write name, content
+  create_uncommitted_file file_name: name, file_content: content
```

Comment:
Looking at this again, I think you can shorten the parameter names a bit. The method name makes it pretty clear that this is about files.

```ruby
create_uncommitted_file name: name, content: content
```

What do you think?